### PR TITLE
Fix bug with installing developer docs.

### PR DESCRIPTION
**Pull request by chrisdembia**
Previously, in the installation directory, both the user and dev doxygen
folders held the user doxygen. With this PR, the dev doxygen is properly
installed into the html_developer folder.

This fixes an issue with the myosin PR doxygen uploads, which incorrectly were showing the user docs.

### CODE REVIEW

File: doc/CMakeLists.txt [hunk 1]
```diff
@@ -182,6 +182,9 @@ if(${DOXYGEN_FOUND})
         list(APPEND DOXYGEN_TARGET_DEPENDENCIES "${html_images_binary_dir}")
 
     endforeach()
+    # Unset loop variables to avoid bugs later on.
+    unset(html_binary_dir)
```
Comment (sherm1) [inline doc/CMakeLists.txt]: is this variable still used somewhere?

Comment (chrisdembia) [inline doc/CMakeLists.txt]: It's used above this line but it's no longer used below this line.

Comment (sherm1) [inline doc/CMakeLists.txt]: OK. The comment and unset are a little, er ... unsettling though. We don't usually do that at the end of a variables useful lifetime so it makes me wonder why it's being done here. I'm fine with it like this if you think it should be there though.

**Comment by sherm1** [comment]
/review. OK to merge.

**Comment by chrisdembia** [comment]
I removed the `unset`s. Thanks for reviewing.

**Pull request merged**